src/services/impl/WindowsClipboard.hh

```diff
@@ -0,0 +1,47 @@
+#ifndef RA_SERVICES_WINDOWS_CLIPBOARD_HH
+#define RA_SERVICES_WINDOWS_CLIPBOARD_HH
+#pragma once
+
+#include "services\IClipboard.hh"
+
+namespace ra {
+namespace services {
+namespace impl {
+
+class WindowsClipboard : public ra::services::IClipboard
+{
+public:
+    void SetText(const std::wstring& sValue) const override
+    {
+        // allocate memory to be managed by the clipboard
+        SIZE_T nSize = (sValue.length() + 1) * 2; // 16-bit characters, with null terminator
+        HGLOBAL hMem = GlobalAlloc(GMEM_MOVEABLE, nSize);
+        if (hMem)
+        {
+            LPVOID pMem = GlobalLock(hMem);
+            if (pMem)
+            {
+                memcpy(pMem, sValue.c_str(), nSize);
+                GlobalUnlock(hMem);
+            }
+
+            // assign the memory to the clipboard
+            if (OpenClipboard(nullptr))
+            {
+                EmptyClipboard();
+                SetClipboardData(CF_UNICODETEXT, hMem);
+                CloseClipboard();
+            }
+            else
+            {
+                GlobalFree(hMem);
```

Comment:
If you aren't gonna make hMem a const pointer I think you should set it to nullptr after freeing its memory to prevent pointer dangling.

Reply:
This seems like a weak argument. If I make it const, it'll still dangle. Additionally, there isn't any code after this line before it goes out of scope.

I'll make it const to match the changes made for`pMem`.